### PR TITLE
First draft to see how it goes

### PR DESCRIPTION
**Pull request by Baggerone**
Trying to take one step at a time.
Am I on the right track?

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  "Version": "2012-10-17",
+  "Statement": [
+        {
+            "Sid": "AllowSESPuts",
+            "Effect": "Allow",
+            "Principal": {
+                "Service": "ses.amazonaws.com"
+            },
+            "Action": "s3:PutObject",
+            "Resource": "arn:aws:s3:::${var.s3_email_bucket/*",
```
Comment (briskt) [inline main.tf]:
```suggestion
            "Resource": "arn:aws:s3:::${var.s3_email_bucket}",
```

File: main.tf [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+}
+
+resource "aws_lambda_function" "ses_redirecter" {
+  s3_bucket        = var.function_bucket_name
+  s3_key           = var.function_zip_name
+  source_code_hash = data.http.function-checksum.body
+  function_name    = var.function_name
+  handler          = "lambda_handler"
+  memory_size      = var.memory_size
+  role             = aws_iam_role.lambdaFunctionRole.arn
```
Comment (briskt) [inline main.tf]:
```suggestion
  role             = aws_iam_role.iam_for_lambda.arn
```

File: main.tf [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+}
+EOF
+
+}
+
+resource "aws_lambda_function" "ses_redirecter" {
+  s3_bucket        = var.function_bucket_name
+  s3_key           = var.function_zip_name
+  source_code_hash = data.http.function-checksum.body
+  function_name    = var.function_name
```
Comment (briskt) [inline main.tf:78]: `function_name` is an undeclared variable

**Review by briskt — APPROVED** [review]
Looks fine to me, just hit a couple snags on `terraform init` and `terraform plan`. Might also want to lock in provider versions.

File: main.tf (hunk 3 above)
Comment (Baggerone) [reply main.tf:78]: :+1:

File: main.tf (hunk 2 above)
Comment (Baggerone) [reply main.tf]: :+1:

File: README.md [hunk 4]
```diff
@@ -1,2 +1,6 @@
 # ses-redirecter-terraform
-Terraform workspace to create an AWS ses-s3-lambda email redirecter environment.
+Terraform workspace to create an AWS SES-S3-Lambda email redirecter environment.
+
+Inspired by [this](https://aws.amazon.com/blogs/messaging-and-targeting/forward-incoming-email-to-an-external-destination/)
+
+Ensure the python code for the lambda exists as a zip file in "${var.function_bucket_name}/${var.function_zip_name}""
```
Comment (fillup) [inline README.md]:
```suggestion
Ensure the python code for the lambda exists as a zip file in "${var.function_bucket_name}/${var.function_zip_name}"
```

**Review by fillup — APPROVED** [review]

**Comment by fillup** [comment]
This looks like a great start. When the kinks are worked out it'd be good to republish this as a terraform module in the official registry for easier reuse and sharing.

**Pull request merged**